### ISSUE TITLE
clustering graphs using more information than just the adjacency matrix?

### ISSUE THREAD

**Issue by edunnwe1** [Statistical-Connectomics-Sp15/intro#94]
Perhaps we touched on this briefly in class, but it's still not quite clear to me: wouldn't there be a limitation in clustering only on the adjacency matrix and ignoring vertex or graph attributes? If we looked at neuronal circuitry and only clustered based on whether neurons synapsed on one another and ignored the properties of the cells, it seems that we might oversimplify/under cluster or incorrectly cluster the system: for example rods can hijack a lot of cone circuitry but we'd want to cluster them as distinct from cones. Computationally, how would you feed more information into a clustering algorithm to improve the result, like clustering on V x E?

**Comment by mblohr** [comment]
The SBM clustering of an adjacency matrix can be a function of latent variables, e.g., vertex attributes, graph attributes, etc.  For example, one quadrant of the matrix can be located in one part of the brain or have directional connections or certain cell properties, versus another quadrant of the matrix, etc.  That's one way vertex, graph, and edge attributes can be folded into clustering via an adjacency matrix.

**Comment by mrjiaruiwang** [comment]
Maybe you could add another field that holds prior information. Say you have access to multiple analysis before starting this one and have probabilities of each being a certain cluster. This could serve as a prior probability and then you could update new labels using baye's rule.
